src/main/cpp/util/path_posix.cc

```diff
@@ -71,12 +69,20 @@ std::string MakeAbsolute(const std::string &path) {
 
 std::string ResolveEnvvars(const std::string &path) {
   std::string result = path;
-  static std::regex env("\\$\\{([^}]+)\\}");
-  std::smatch m;
-  while (std::regex_search(result, m, env)) {
-    const char *value = getenv(m[1].str().c_str());
-    const std::string replacement = std::string(value ? value : "");
-    result.replace(m.position(0), m.length(0), replacement);
+  size_t start = 0;
+  while ((start = result.find("${", start)) != std::string::npos) {
+    // Just match to the next }
+    size_t end = result.find("}", start + 1);
+    if (end == std::string::npos) {
+      // No more variables can exist
+      break;
```

Comment:
Shouldn't this fail? It is a malformed name.

Reply:
I suppose it can, I'll reflect the BAZEL_DIE behaviour from the windows code